typed-racket-lib/typed-racket/types/pairwise-intersect.rkt

```diff
@@ -0,0 +1,121 @@
+#lang racket
+
+;; This module provides a function pairwise-intersect for calculating a lower
+;; bound of two types with respect to the precision order (*not* the subtype
+;; order). Contract typechecking uses this function to determine the type of a
+;; contracted value, given the value's type and the contract's output type.
+
+(require "../utils/utils.rkt"
+         (rep type-rep prop-rep object-rep values-rep)
+         (types subtype)
+         (only-in (infer infer) intersect)
+         racket/match)
+
+(provide pairwise-intersect)
+
+(define (pairwise-intersect/arr s t)
+  (match* (s t)
+    [((Arrow: s-dom s-rest s-kws s-rng)
+      (Arrow: t-dom t-rest t-kws t-rng))
+     #:when (= (length s-kws) (length t-kws))
+     (make-Arrow (map pairwise-intersect s-dom t-dom)
+                 (and s-rest t-rest
+                      (pairwise-intersect s-rest t-rest))
+                 (map pairwise-intersect/kw s-kws t-kws)
+                 (pairwise-intersect s-rng t-rng))]
+    [(_ _)
+     (raise-arguments-error
+      'pairwise-intersect/arr
```

Comment:
`int-err` raises internal TR errors (we usually use that or just omit an "else" match case.